### ISSUE TITLE
Windows don't retain their placement if made fullscreen from within the application

### ISSUE THREAD

**Issue by simtd** [leftwm/leftwm#1102]
### Describe the bug

When making an application fullscreen from within the application itself  it changes placement in the stack to the top when un-fullscreened again. I.e. you're not using ToggleFullScreen but rather clicking a button inside the app to make it fullscreen. This commonly happens with video players like youtube or mpv.

https://github.com/leftwm/leftwm/assets/69794006/e337b13d-ad18-47fe-819e-85dc08fb2958

### Steps to reproduce (optional)

The easiest way to reproduce this is by watching a youtube video in your browser:
1) Have the browser window down in the stack somewhere
2) Fullscreen and un-fullscreen the youtube video by double tapping it or clicking the icon in the bottom right
3) The browser windows should now be at the top of the stack

### OS / Distro

Arch

### LeftWM Check

```text
:: LeftWM version: 0.4.2
:: LeftWM git hash: 357aea5
:: Enabled features:  journald-log lefthk
:: Checking feature dependencies . . .
    -> journald-log OK
    -> lefthk OK
:: Checking for leftwm binaries . . .
    -> Binaries OK
:: Loading configuration . . .
    -> Configuration loaded OK
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK
:: Checking theme . . .
    -> Theme OK
```

**Comment by VuiMuich** [comment]
:thinking: I guess #1052 overlooked the situation where the window itself demands fullscreen.
I'd say this should not be too hard to move the logic of storing the window order and restoring it from the command handler to a more generic place and trigger also on `xlib` events.
Might be a good introduction to our code base, so I'll add the `good first issue` label.

**Comment by simtd** [comment]
Btw you can trigger this with fullscreen youtube videos, that's the easiest way.

**Comment by barkyq** [comment]
It works as expected for me; the window preserves its position in the layout after turning on and off fullscreen using the youtube player (I am using brave browser). Does it still not work with the current version?

:: LeftWM version: 0.5.1
:: LeftWM git hash: 0.5.1-15-g50d5333
:: Enabled features:  journald-log lefthk
:: Checking feature dependencies . . .
    -> journald-log OK
    -> lefthk OK
:: Checking for leftwm binaries . . .
    -> Binaries OK
:: Loading configuration . . .
    -> Configuration loaded OK
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK
:: Checking theme . . .
    -> Theme OK

**Comment by nemalex** [comment]
It works for me too
I think it was fixed in a recent version

**Comment by Eskaan** [comment]
This was initially implemented in #1052 and really fixed as a side-effect of #1156 in #1158 where I stopped to re-assign the whole window stack each time we sorted it.

To be exact, the issue was this single line, fixed by removing it: https://github.com/leftwm/leftwm/blob/91c506aa712200f242d4226f233cd44bee61db71/leftwm-core/src/state.rs#L109
